lib/pbench/test/functional/server/test_put.py

```diff
@@ -106,3 +130,24 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
         assert count == len(
             self.tarballs
         ), f"Didn't find all expected datasets, found {count} of {len(self.tarballs)}"
+
+    def test_delete(self, server_client: PbenchServerClient, login_user):
+        """Verify we can delete each previously uploaded dataset."""
+        datasets = server_client.get_list()
+        datasets_hash = {}
+        try:
+            for dataset in datasets:
+                datasets_hash[f"{dataset.name}.tar.xz"] = dataset.resource_id
+        except HTTPError as exc:
+            pytest.fail(
+                f"Unexpected HTTP error, url = {exc.response.url}, status code = {exc.response.status_code}, text = {exc.response.text!r}"
+            )
+        for t in TARBALL_DIR.glob("*.tar.xz"):
+            resource_id = datasets_hash.get(t.name)
+            assert resource_id, f"Expected to find tar ball {t.name} to delete"
+            response = server_client.post(
+                api=API.DATASETS_DELETE,
+                uri_params={"dataset": resource_id},
+                raise_error=False,
+            )
```

Comment:
Since the intent of the client library was to make API use easy, especially with regard to API templating, I'd prefer to encapsulate this in a `delete_dataset` method analogous to `upload` (which I suppose should probably have been `upload_dataset` even though we currently don't "upload" anything else).

Reply:
Do you want me to make this change to the client API as part of these tests?

Reply:
I would vote for doing the expansion now, while we are thinking about it and know what we want.  If we put it off there won't be much impetus to do it, and we'll end up with client code spread around....

Reply:
I'd rather not expand a PR beyond its intended purpose.  The goal is to get the tests running under `tox` and make the tests works against an existing server so that they leave no trace of their execution behind.

I believe this PR does that.

I am happy to work on another PR which expands the client interface.

Reply:
> I'd rather not expand a PR beyond its intended purpose.

That's fair.  So, the appropriate way to proceed would be to implement the expanded client interface which this PR needs, and then base this PR on that change....